src/data_structures/synchronizer.ml

```diff
@@ -0,0 +1,64 @@
+(* SPDX-License-Identifier: AGPL-3.0-or-later *)
+(* Copyright © 2021-2024 OCamlPro *)
+(* Written by the Owi programmers *)
+
+type ('get, 'write) t =
+  { mutex : Mutex.t
+  ; cond : Condition.t
+  ; getter : unit -> 'get option
```

Comment:
It is not clear to me why this has type `unit -> 'get option` and not `'get option` by looking at the code, could you explain ?

Reply:
The idea is that this is a closure that when called, gets one element to run, without caring for synchronization. Typically, for a Queue, this is `fun () -> Queue.take_opt q`, and for concolic this will be `fun () -> find_node_to_run tree`.